modules/data.remote/R/call_MODIS.R

```diff
@@ -97,40 +152,71 @@ call_MODIS <- function(outfolder = ".", start_date, end_date, lat, lon, size = 0
     {
       SD <- rep("nan", nrow(dat))
     } else {
-      SD <- as.numeric(sd$value) * as.numeric(sd$scale) #formatC(sd$data$data*scale, digits = 2, format = 'f')
+      SD <- as.numeric(sd$value) * as.numeric(sd$scale) 
     }
     
-    output <- as.data.frame(cbind(dat$modis_date, dat$calendar_date, dat$band, dat$tile, dat$latitude, dat$longitude, dat$pixel, dat$value, QC, SD), stringsAsFactors = F)
+    output <- as.data.frame(cbind(dat$modis_date, dat$calendar_date, dat$band, dat$tile, dat$latitude, dat$longitude, dat$pixel, dat$value, QC, SD), stringsAsFactors = FALSE)
     names(output) <- c("modis_date", "calendar_date", "band", "tile", "lat", "lon", "pixels", "data", "qc", "sd")
     
-    output[,5:10] <- lapply(output[,5:10], as.numeric)
+    output[ ,5:10] <- lapply(output[,5:10], as.numeric)
     
     # scale the data + stdev to proper units
     output$data <- output$data * (as.numeric(dat$scale))
     output$sd <- output$sd * (as.numeric(dat$scale))
     output$lat <- round(output$lat, 4)
     output$lon <- round(output$lon, 4)
     
-    fname <- paste(product, "_", band, "output_", start_date, "_", end_date, "_", lat, "_", lon, ".csv", sep = "")
-    fname <- paste0(outfolder, "/", fname)
-    write.csv(output, fname)
-    return(output)}
-
+    if (QC_filter)
+    {
+      output$qc == as.character(output$qc)
+      for (i in seq_len(nrow(output)))
+      {
+        convert <- paste(binaryLogic::as.binary(as.integer(output$qc[i]), n = 8), collapse = "")
+        output$qc[i] <- substr(convert, nchar(convert)-2, nchar(convert))
+      }
+      good <- which(output$qc %in% c("000", "001"))
+      if (length(good) > 0 || !(is.null(good)))
+      {
+        output = output[good, ]
+      } else {
+        print("All QC values are bad. No data to output with QC filter == TRUE.")
+      }
+    }
+    
+   
+    if (!(outfolder == ""))
+    {
+      if (!(is.null(siteID)))
+      {
+        fname <- paste(product, "_", band, "_", siteID, "_output_", start_date, "_", end_date, ".csv", sep = "")
+      } else {
+        fname <- paste(product, "_", band, "_output_", lat, "_", lon, "_", start_date, "_", end_date, ".csv", sep = "")
+      }
+      fname <- file.path(outfolder, fname)
+      write.csv(output, fname, row.names = FALSE)
+    }
+    
+    return(output)
+  }
+  
   
   if (package_method == "reticulate"){
     # load in python script
     script.path <- file.path(system.file("extract_modis_data.py", package = "PEcAn.data.remote"))
-    #script.path = file.path('/Users/bmorrison/pecan/modules/data.remote/inst/extract_modis_data.py')
     reticulate::source_python(script.path)
     
     # extract the data
     output <- extract_modis_data(product = product, band = band, lat = lat, lon = lon, start_date = start_date, end_date = end_date, size = size, band_qc = band_qc, band_sd = band_sd)
-    output[,5:10] <- lapply(output[,5:10], as.numeric)
+    output[ ,5:10] <- lapply(output[ ,5:10], as.numeric)
```

Comment:
do we need these spaces?

Reply:
yea. I probably missed some. I'll double check for missing spaces